src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -3247,6 +3297,22 @@ - (void)removeClientDataForKey:(NSString *)key endpointID:(NSNumber *)endpointID
     [self.temporaryMetaDataCache removeObjectForKey:[NSString stringWithFormat:@"%@:%@", key, endpointID]];
 }
 
+#pragma mark Log Help
+
+- (MTRDeviceEssentialAttributes *)_essentialAttributesForCurrentState {
+    MTRClusterPath * basicInfoClusterPath = [MTRClusterPath clusterPathWithEndpointID:@(kRootEndpointId) clusterID:@(MTRClusterIDTypeBasicInformationID)];
+    MTRDeviceClusterData * basicInfoClusterData = [self _clusterDataForPath:basicInfoClusterPath];
+    
+    NSNumber * vidObj = basicInfoClusterData.attributes[@(MTRAttributeIDTypeClusterBasicInformationAttributeVendorIDID)][MTRValueKey];
```

Comment:
This is not safe.  If the server happened to server non-spec-compliant TLV we could have some other type of object for the value, and now you get bad behavior.

I strongly suggest using `[[MTRAttributeReport alloc] initWithResponseValue:...]` which for attributes we know about (which these are) will verify that the types are correct and put the value (In this case `NSNumber *`) into the `.value` of the `MTRAttributeReport`.